packages/hydrogen/src/routing/graphiql.ts

```diff
@@ -1,21 +1,60 @@
 import type {LoaderFunctionArgs} from '@remix-run/server-runtime';
-import type {Storefront} from '../storefront';
 
 type GraphiQLLoader = (args: LoaderFunctionArgs) => Promise<Response>;
 
 export const graphiqlLoader: GraphiQLLoader = async function graphiqlLoader({
-  context,
+  request,
+  context: {storefront, customerAccount},
 }: LoaderFunctionArgs) {
-  const storefront = context?.storefront as Storefront | undefined;
+  const url = new URL(request.url);
+
   if (!storefront) {
     throw new Error(
       `GraphiQL: Hydrogen's storefront client must be injected in the loader context.`,
     );
   }
 
-  const url = storefront.getApiUrl();
-  const accessToken =
-    storefront.getPublicTokenHeaders()['X-Shopify-Storefront-Access-Token'];
+  const schemas: {
+    [key: string]: {
+      name: string;
+      value?: object;
+      accessToken?: string;
+      authHeader: string;
+      apiUrl: string;
+    };
+  } = {};
+
+  if (storefront) {
+    const authHeader = 'X-Shopify-Storefront-Access-Token';
+    schemas.storefront = {
+      name: 'Storefront API',
+      authHeader,
+      accessToken: storefront.getPublicTokenHeaders()[authHeader],
+      apiUrl: storefront.getApiUrl(),
+    };
+  }
+
+  if (customerAccount) {
+    // CustomerAccount API does not support introspection to the same URL.
+    // Read it from a file using the asset server:
```

Comment:
@s-lee-kwong Can we add introspection to the graphql end point?

Reply:
It is something we can look at but right now our only introspection point is something like

```https://app.myshopify.com/services/graphql/introspection/customer?api_client_api_key=159a99b8a7289a72f68603f2f4de40ac&api_version=2024-01```

but that api_key is from an app so I wouldn't use what I pasted exactly.

Reply:
GraphiQL can only work with 1 url for both introspection and queries. So if you want to add support for GraphiQL in CAAPI, the URL for queries should also provide introspection 🙏 

Here we had to work around this issue by downloading the schema in a different way and providing it to GarphiQL in JSON format without doing introspection.

Reply:
I will add this to the on going Friction Doc for the future. 

Having the introspection and queries be at the same URL is a fairly standard way of implementation for GraphQL API. A lot of the tools work assuming this convention and it would be really good to have.